src/main/kotlin/com/ex_dock/ex_dock/database/product/ProductCustomAttributesJdbcVerticle.kt

```diff
@@ -0,0 +1,196 @@
+package com.ex_dock.ex_dock.database.product
+
+import com.ex_dock.ex_dock.database.connection.Connection
+import io.vertx.core.AbstractVerticle
+import io.vertx.core.eventbus.EventBus
+import io.vertx.core.json.JsonObject
+import io.vertx.kotlin.core.json.json
+import io.vertx.kotlin.core.json.obj
+import io.vertx.sqlclient.Pool
+import io.vertx.sqlclient.Row
+import io.vertx.sqlclient.Tuple
+
+class ProductCustomAttributesJdbcVerticle: AbstractVerticle() {
+  private lateinit var client: Pool
+  private lateinit var eventBus: EventBus
+  private val failedMessage: String = "failed"
+
+  override fun start() {
+    client = Connection().getConnection(vertx)
+    eventBus = vertx.eventBus()
+
+    getAllCustomAttributes()
+    getCustomAttributeByKey()
+    createCustomAttribute()
+    updateCustomAttribute()
+    deleteCustomAttribute()
+  }
+
+  private fun getAllCustomAttributes() {
+    val getAllCustomAttributesConsumer = eventBus.consumer<String>("process.attributes.getAllCustomAttributes")
+    getAllCustomAttributesConsumer.handler { message ->
+      val query = "SELECT * FROM custom_product_attributes"
+      val rowsFuture = client.preparedQuery(query).execute()
+      var json: JsonObject
```

Comment:
As privately discussed, I don't think it is necessary to pick this up at this point in time since a lot of it would've needed to be encoded in JSON anyway.

Reply:
We found a generic codec that we can use that made an almost 8% speed improvement in our tests